### PR TITLE
[WIP] Add snapcast control

### PR DESCRIPTION
**Pull request by AlexProgrammerDE**
Change-type: minor

### CODE REVIEW

**Comment by AlexProgrammerDE** [comment]
@tmigone is there a way to get the ip of the master server in other containers?

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by AlexProgrammerDE** [comment]
@tmigone i am almost finshed here. Can you plese help me with the comment above?

**Comment by tmigone** [comment]
Since the container is using `network_mode: host` you can get it from `http://localhost:3000`.
Also, can you briefly tell me the benefits of using `snapcastrd`?

**Comment by AlexProgrammerDE** [comment]
@tmigone [snapcastr](https://github.com/xkonni/snapcastr) is a webinterface for controlling a snapcast server. You can control streams and the sound volume for each client conected to the server. Creating zones and groups is also possible.

**Comment by AlexProgrammerDE** [comment]
@tmigone this could help to create a web front-end. Do you think we should create a web dashboard for balena?

**Comment by tmigone** [comment]
@AlexProgrammerDE interesting! Going to try it over the weekend!

About the web dashboard, I don't think at this point it's needed. Specially if this snapcastrd interface can solve some problems (namely volumes and such) I think I'd first see what we can do with it.

**Comment by frederickjansen** [comment]
@tmigone Is the idea to make this part of the 3.0 roadmap? Or are you envisioning going with a different approach?

**Comment by tmigone** [comment]
@AlexProgrammerDE @frederickjansen having a local UI where you can control project parameters is definitely something we want to have in 3.x. This local UI concept though ties in to some platform features we are working on that are not ready yet. Basically this problem applies to any balena project so we want something that works for all of them, not only sound, so I think this should not be an immediate focus (not 3.0, maybe a minor version afterwards 3.1/3.2).

This `snapcastrd` based approach seems good enough for now, so I think we can have it for 2.x. I'll make sure I check it out asap (i know i've said that already 😬 )

**Comment by AlexProgrammerDE** [comment]
@balena-ci retest

**Comment by tmigone** [comment]
@balena-ci rebase

**Comment by tmigone** [comment]
Hey @AlexProgrammerDE I had to rebase/squash manually this PR, balena-CI couldn't do it because of all the merges from master it had. 

As a general rule it's better not to use the "Update branch" button (main reason being that you get all these commits that say "merge master bla bla", instead if you need to bring changes from master into your PR branch you can use:
- "(at)balena-ci rebase", if there are are no conflicts it should auto-rebase. 
- If there are conflicts (because you are changing stuff that already changed in master) then you'll need to rebase using the git cli.

File: snapcast-control/start.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+snapcastrd --port=80 --sc_host=localhost
```
Comment (tmigone) [inline snapcast-control/start.sh:3]: To get the IP address of the current master server you can do:

```
  while ! curl -s "http://localhost:3000"; do sleep 1; done
  SNAPCAST_SERVER=$(curl --silent http://localhost:3000)
snapcastrd --port=80 --sc_host "$SNAPCAST_SERVER"
```

For the full example checkout: https://github.com/balenalabs/balena-sound/blob/master/snapcast-client/start.sh

Comment (tmigone) [reply snapcast-control/start.sh:3]: Also, after `fleet-supervisor` [Line27](https://github.com/balenalabs/balena-sound/blob/master/fleet-supervisor/server.js#L27) you'll need to restart this service with:

```
restartBalenaService('snapcast-control')
```

so each time the master server changes the `snapcast-control` finds out too.

File: snapcast-control/Dockerfile.template [hunk 2]
```diff
@@ -0,0 +1,26 @@
+FROM balenalib/%%BALENA_MACHINE_NAME%%-debian:buster
```
Comment (tmigone) [inline snapcast-control/Dockerfile.template:1]: Maybe we should try multi-stage dockerfile or base it on alpine to try and reduce image size.

**Comment by tmigone** [comment]
@AlexProgrammerDE I left a few comments. It's cool that it works, but I feel like the only real functionality we can take advantage at this point is the volume controls, right? Having zones would be cool, but as far as I read we need multiple fifo streams for that, and while we could do it i feel like we will create a mess of env vars. I'll try to read more about this.

**Comment by AlexProgrammerDE** [comment]
imma close this for now. I think you already implemented a secret ui that allows us to change the volume. 👍

**Pull request closed without merging**